src/main/java/com/netflix/imflibrary/st0377/HeaderPartition.java

```diff
@@ -637,7 +637,7 @@ public BigInteger getEssenceDuration(){
 
             for(InterchangeObject.InterchangeObjectBO interchangeObjectBO : structuralComponentBOs){
                 StructuralComponent.StructuralComponentBO structuralComponentBO = (StructuralComponent.StructuralComponentBO) interchangeObjectBO;
-                duration += structuralComponentBO.getDuration();
```

Comment:
This summation is needed. However real issue is that the duration is not reset for each timeline track. 
Thanks for pointing out the issue with timeline track duration calculation. I have included the correct fix here: 
 https://github.com/Netflix/photon/pull/214

Reply:
Thank you.